infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/TestSnapshotReader.java

```diff
@@ -40,23 +40,24 @@ public SnapshotReadMessage read(UUID snapshotRequestId) {
         // Connect to endpoint
         List<LogReplicationEntry> messages = new ArrayList<>();
 
+        int index = globalIndex;
+
+        // Limit to read as max as BatchSize and until the maximum baseSnapshot
+        for (int i=index; (i<(index+config.getBatchSize()) && index<baseSnapshot) ; i++) {
```

Comment:
index is never getting updated.  will it ever reach the condition index<baseSnapshot?

Reply:
it is in line 43